### PR TITLE
cataclysm-dda-git: remove incompatible patches, bump git

### PR DESCRIPTION
**Pull request by caryoscelus**
## Description of changes

update git variety of cataclysm-dda by removing patches that are no longer applicable (and hopefully not required)

note that i've actually done this while working on adding Cataclysm: Bright Nights (a fork of C:DDA) package. if this PR is not going to be reviewed/merged i'm likely just going to fuse this into upcoming PR for that

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/games/cataclysm-dda/git.nix
Comment (DeeUnderscore) [inline pkgs/games/cataclysm-dda/git.nix:27]: Not sure this one is safe to remove. If the Nixpkgs Darwin+tiles build still puts translation files in `share/locale` (by running `make install` during install phase), then we'll need that patch for the resulting binary to pick the translation files up.

**Review by DeeUnderscore — CHANGES_REQUESTED** [review]

File: pkgs/games/cataclysm-dda/git.nix
Comment (caryoscelus) [reply pkgs/games/cataclysm-dda/git.nix:27]: ah, looks like i confused translations.cpp with translation.cpp, thought the corresponding lines were removed completely. my bad

**Comment by rardiol** [comment]
Seems to be working fine on NixOS, can't check the Darwin issue myself.

File: pkgs/games/cataclysm-dda/git.nix
Comment (rardiol) [reply pkgs/games/cataclysm-dda/git.nix:27]:
```
diff --git a/src/translations.cpp b/src/translations.cpp
--- a/src/translations.cpp
+++ b/src/translations.cpp
@@ -52,13 +52,11 @@ std::string locale_dir()
 #define CATA_IS_ON_BSD
 #endif

-#if !defined(__ANDROID__) && ((defined(__linux__) || defined(CATA_IS_ON_BSD) || (defined(MACOSX) && !defined(TILES))))
     if( !PATH_INFO::base_path().empty() ) {
         loc_dir = PATH_INFO::base_path() + "share/locale";
     } else {
         loc_dir = PATH_INFO::langdir();
     }
-#endif
 #endif // LOCALIZE
     return loc_dir;
 }
```

Patch for the git version of the package.

**Review by DeeUnderscore — CHANGES_REQUESTED** [review]
The remaining patches look to already be in master, so it looks good, but please squash down to a single commit.

**Review by DeeUnderscore — APPROVED** [review]
Great, thank you!

**Comment by caryoscelus** [comment]
@DeeUnderscore done

**Review by rardiol — APPROVED** [review]

**Comment by caryoscelus** [comment]
@rardiol @DeeUnderscore would you say it would make sense for Bright Nights fork package to be included (lets say as bn.nix and bn-git.nix) in the same directory or keep them separate? so far it seems the only compilation change required (besides different source) `-Wno-ignored-attributes` flag

**Comment by rardiol** [comment]
On Tuesday, 30 July 2024 17.14.23 GMT-3 caryoscelus wrote:
> @rardiol @DeeUnderscore would you say it would make sense for Bright Nights
> fork package to be included (lets say as bn.nix and bn-git.nix) in the same
> directory or keep them separate? so far it seems the only compilation
> change required (besides different source) `-Wno-ignored-attributes` flag
If it is a fork and is essentially the same derivation and both will evolve 
together I think it makes sense to keep both in the same directory.

**Comment by cizra** [comment]
Tested with Linux, works fine. Could we merge this PR separately, now, to get a baseline working, and then continue working on C:BN?

**Comment by caryoscelus** [comment]
@cizra that was what i expected

**Review by cizra — APPROVED** [review]
Waiting for someone with merge access, then... I wonder if there's a procedure to get their attention.

**Comment by DeeUnderscore** [comment]
>  Waiting for someone with merge access, then... I wonder if there's a procedure to get their attention.

There is a thread on the Discourse that can be used for that, where I've posted this (see previous automated comment)

**Pull request merged**
